build/parseSpec.c

```diff
@@ -1129,3 +1157,26 @@ rpmSpec rpmSpecParse(const char *specFile, rpmSpecFlags flags,
 {
     return parseSpec(specFile, flags, buildRoot, 0);
 }
+
+rpmRC parseGeneratedSpecs(rpmSpec spec)
+{
+    ARGV_t argv = NULL;
+    int argc = 0;
+    int i;
+
+    char * specPattern = rpmGenPath("%{specpartsdir}", NULL, "*.specpart");
+    if (rpmGlob(specPattern, &argc, &argv) == 0) {
+	for (i = 0; i < argc; i++) {
+	    rpmlog(RPMLOG_NOTICE, "Reading %s\n", argv[i]);
+	    pushOFI(spec, argv[i]);
+	    snprintf(spec->fileStack->readBuf, spec->fileStack->readBufLen,
+		     "# Spec part read from %s\n\n", argv[i]);
+	    if (parseSpecSection(spec, 1)) {
```

Comment:
This should've probably been `!parseSpecSection(spec, 1)`

Reply:
Yup. Since parseSpecSection() doesn't actually allocate the spec structure, it should return a simple 0/1 error code instead of a pointer.

Reply:
> Do stuff for others to clean up is not an attitude us maintainers can afford you know.

Yes, I know. I do not actually expect anyone to clean this up. I expect this to stay that way as long as we have BUILDARCH - which is forever.